### PR TITLE
[bitnami/elasticsearch] Fix failed metrics pod in elasticsearch when deploy ES

### LINKED ISSUE #12957: Pod metrics in elasticsearch failed when deploy elasticsearch
### Name and Version

bitnami/elasticsearch 19.4.2

### What steps will reproduce the bug?
```
helm install --wait -n elasticsearch my-elasticsearch bitnami/elasticsearch -f value.yaml
```

### Are you using any custom parameters or values?
```
master:
  replicaCount: 1
  resources:
    limits:
      cpu: 500m
      memory: 2Gi
    requests:
      cpu: 500m
      memory: 256Mi

data:
  replicaCount: 1
  resources:
    limits:
      cpu: 500m
      memory: 2Gi
    requests:
      cpu: 500m
      memory: 256Mi

coordinating:
  replicaCount: 1
  resources:
    limits:
      cpu: 500m
      memory: 2Gi
    requests:
      cpu: 500m
      memory: 256Mi

metrics:
  enabled: true
```

### What is the expected behavior?

install elasticsearch without error

### What do you see instead?

https://gist.github.com/patsevanton/91ff1d968f708e3731fa36330a2c8b13

### Additional information
```
kubectl get pod -n elasticsearch
NAME                                       READY   STATUS    RESTARTS   AGE
my-elasticsearch-coordinating-0            1/1     Running   0          22m
my-elasticsearch-data-0                    1/1     Running   0          22m
my-elasticsearch-ingest-0                  1/1     Running   0          22m
my-elasticsearch-ingest-1                  1/1     Running   0          22m
my-elasticsearch-master-0                  1/1     Running   0          22m
my-elasticsearch-metrics-944c9685b-bpwr5   1/1     Running   1          22m
```

### PR DESCRIPTION
**Pull request by patsevanton**
Signed-off-by: Anton Patsev <patsev.anton@gmail.com>

### Description of the change
Fix failed metrics pod in elasticsearch when deploy elasticsearch

### Benefits

Fixed bug https://github.com/bitnami/charts/issues/12957
### Possible drawbacks

Limitations: run metrics after run core pods elasticsearch
### Applicable issues

  - fixes #12957

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/elasticsearch/templates/metrics/deployment.yaml [hunk 1]
```diff
@@ -11,9 +11,11 @@ metadata:
...
     {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
     {{- end }}
-  {{- if .Values.commonAnnotations }}
-  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
-  {{- end }}
+  annotations:
+    {{- include "common.tplvalues.render" ( dict "value" .Values.metrics.annotations "context" $ ) | nindent 4 }}
+      {{- if .Values.commonAnnotations }}
+      {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+      {{- end }}
```
Comment (jotamartos) [inline bitnami/elasticsearch/templates/metrics/deployment.yaml:18]: Please include a new if block to ensure the values exist and use 4 spaces in all lines 

```
  annotations:
    {{- if .Values.metrics.annotations }}
    {{- include "common.tplvalues.render" (dict "value" .Values.metrics.annotations "context" $) | nindent 4 }}
    {{- end }}
    {{- if .Values.commonAnnotations }}
    {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
    {{- end }}
```

**Review by jotamartos — CHANGES_REQUESTED** [review]

File: bitnami/elasticsearch/templates/metrics/deployment.yaml (hunk 1 above)
Comment (patsevanton) [reply bitnami/elasticsearch/templates/metrics/deployment.yaml]: Thanks for advice. Added.

Comment (jotamartos) [reply bitnami/elasticsearch/templates/metrics/deployment.yaml]: Sorry, I can still see that there are extra spaces in the lines you included. Please use just 4 spaces at the end so everything has the same number of spaces in the template

```
  annotations:
    {{- if .Values.metrics.annotations }}
    {{- include "common.tplvalues.render" (dict "value" .Values.metrics.annotations "context" $) | nindent 4 }}
    {{- end }}
    {{- if .Values.commonAnnotations }}
    {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
    {{- end }}
```

**Comment by patsevanton** [comment]
@jotamartos Hello
New Pull request https://github.com/bitnami/charts/pull/13303
I am sorry. This pull request have merge conflict. I push to sync fork. I created new PR.

**Pull request closed without merging**